### PR TITLE
improvement: Allow Tags To Be Applied To Public/Private Subnets Based Off Of Their Names

### PR DESCRIPTION
**Pull request by toneill818**
## Description
Added two new variables `private_subnet_tags_per_name` and `public_subnet_tags_per_name`. This gives the ability to add tags to a specific subnet. 

## Motivation and Context
Fixes Issue #1129

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -120,7 +120,11 @@ resource "aws_subnet" "public" {
     },
     var.tags,
     var.public_subnet_tags,
-    lookup(var.public_subnet_tags_per_az, element(var.azs, count.index), {})
+    lookup(var.public_subnet_tags_per_az, element(var.azs, count.index), {}),
+    lookup(var.public_subnet_tags_per_name, try(
+      var.public_subnet_names[count.index],
+      format("${var.name}-${var.public_subnet_suffix}-%s", element(var.azs, count.index))
```
Comment (GeertWille) [inline main.tf:126]: I get errors on this. I tried using ```lookup(var.public_subnet_tags_per_name, try(var.public_subnet_names[count.index], ""), {}),``` which works better imo

File: main.tf [hunk 2]
```diff
@@ -255,7 +259,11 @@ resource "aws_subnet" "private" {
     },
     var.tags,
     var.private_subnet_tags,
-    lookup(var.private_subnet_tags_per_az, element(var.azs, count.index), {})
+    lookup(var.private_subnet_tags_per_az, element(var.azs, count.index), {}),
+    lookup(var.private_subnet_tags_per_name, try(
```
Comment (GeertWille) [inline main.tf:263]: Same here I used ```lookup(var.private_subnet_tags_per_name, try(var.private_subnet_names[count.index], ""), {}),```

**Comment by fatmcgav** [comment]
FWIW, #1139 has passed all CI checks 👍

**Comment by toneill818** [comment]
Closing in favor of PR #1139

**Pull request closed without merging**